panimg/image_builders/metaio_utils.py

```diff
@@ -189,23 +198,45 @@ def validate_and_clean_additional_mh_headers(
     for key, value in headers.items():
         if key in EXPECTED_HEADERS:
             cleaned_headers[key] = value
-        else:
-            if key in ADDITIONAL_HEADERS:
-                match_pattern = ADDITIONAL_HEADERS[key]
-                if not re.match(match_pattern, value):
-                    raise ValueError(
-                        f"Invalid data type found for "
-                        f"additional header key: {key}"
-                    )
-                cleaned_headers[key] = value
-        if key in HEADERS_MATCHING_NUM_TIMEPOINTS:
-            validate_list_data_matches_num_timepoints(
-                headers=headers, key=key, value=value
-            )
+        elif key in ADDITIONAL_HEADERS:
+            match_pattern = ADDITIONAL_HEADERS[key]
+            if not re.match(match_pattern, value):
+                raise ValueError(
+                    f"Invalid data type found for "
+                    f"additional header key: {key}"
+                )
+            if key in HEADERS_MATCHING_NUM_TIMEPOINTS:
+                validate_list_data_matches_num_timepoints(
+                    headers=headers, key=key, value=value
+                )
+            if key in HEADERS_MATCHING_WINDOW_SETTINGS:
+                validate_center_matches_width_setting(headers, key, value)
+            cleaned_headers[key] = value
 
     return cleaned_headers
 
 
+def validate_center_matches_width_setting(
+    headers: Dict[str, str], key: str, value: str
+):
+    window_keys = ("WindowWidth", "WindowCenter")
+    if key not in window_keys:
+        return
+    if not re.match(FLOAT_ARRAY_MATCH_REGEXP, value):
+        return
+
+    counter_key = window_keys[0] if key == window_keys[1] else window_keys[1]
+    if not re.match(FLOAT_ARRAY_MATCH_REGEXP, headers[counter_key]):
+        raise ValueError(
```

Comment:
Some legacy code cleanup may help here. I do not think that ValueError is the correct thing to raise here. Also, it is not caught by the caller (https://github.com/DIAGNijmegen/rse-panimg/blob/73e04d27f872d758517dcfdc4910a9d446087768/panimg/image_builders/metaio_mhd_mha.py#L100) causing the entire import to stop. Maybe this can be updated to a ValidationError, caught and fed back to the user?